src/core/announcements/validation.ts

```diff
@@ -0,0 +1,206 @@
+
+  return true;
+};
+
+/**
+ * isReactionAnnouncement() is a type check for ReactionAnnouncement
+ *
+ * @param obj - The object to check
+ * @returns True if the object is a ReactionAnnouncement, otherwise false
+ */
+const isReactionAnnouncement = (obj: unknown): obj is ReactionAnnouncement => {
+  if (!isRecord(obj)) return false;
+  if (obj["announcementType"] != AnnouncementType.Reaction) return false;
+  if (!isDSNPUserId(obj["fromId"])) return false;
+  if (!isValidEmoji(obj["emoji"])) return false;
+  if (!isDSNPAnnouncementId(obj["inReplyTo"])) return false;
+
+  return true;
+};
+
+/**
+ * isProfileAnnouncement() is a type check for ProfileAnnouncement
+ *
+ * @param obj - The object to check
+ * @returns True if the object is a ProfileAnnouncement, otherwise false
+ */
+const isProfileAnnouncement = (obj: unknown): obj is ProfileAnnouncement => {
+  if (!isRecord(obj)) return false;
+  if (obj["announcementType"] != AnnouncementType.Profile) return false;
+  if (!isDSNPUserId(obj["fromId"])) return false;
+  if (!isString(obj["url"])) return false;
+  if (!isString(obj["contentHash"])) return false;
+
+  return true;
+};
+
+/**
+ * isAnnouncement() is a type check for Announcement
+ *
+ * @param obj - The object to check
+ * @returns True if the object is a Announcement, otherwise false
+ */
+const isAnnouncement = (obj: unknown): obj is Announcement => {
+  if (!isRecord(obj)) return false;
+  if (!isAnnouncementType(obj["announcementType"])) return false;
+
+  const validators = {
+    [AnnouncementType.GraphChange]: isGraphChangeAnnouncement,
+    [AnnouncementType.Broadcast]: isBroadcastAnnouncement,
+    [AnnouncementType.Reply]: isReplyAnnouncement,
+    [AnnouncementType.Reaction]: isReactionAnnouncement,
+    [AnnouncementType.Profile]: isProfileAnnouncement,
+  };
+
+  return validators[obj["announcementType"]](obj);
+};
+
+/**
+ * isSignedAnnouncement() is a type check for SignedAnnouncements
+ *
+ * @param obj - The object to check
+ * @returns True if the object is a SignedAnnouncement, otherwise false
+ */
+export const isSignedAnnouncement = (obj: unknown): obj is SignedAnnouncement => {
+  if (!isRecord(obj)) return false;
+  if (!isString(obj["signature"])) return false;
+
+  return isAnnouncement(obj);
+};
+
+/**
+ * isValidAnnouncement() validates an announcement and its associated content.
+ * Note that this method only validates an announcement's signature, format and
+ * linked activity content. It **does not** validate any audio, image or video
+ * files linked in the activity content object, so these files should be
+ * validated separately.
+ *
+ * @param obj - A signed announcement to validate
+ * @param opts - Optional. Configuration overrides, such as from address, if any
+ * @returns True if the announcement is valid, otherwise false
+ */
+export const isValidAnnouncement = async (obj: unknown, opts?: ConfigOpts): Promise<boolean> => {
+  const fetch = requireGetFetchFunc(opts);
```

Comment:
I'm not sure we want to get into the fetch business. Although we are with the batch Parquet files (as we don't need to fetch the entire content).

There are just so few times that I can imagine that someone would want to call this function and not either have or need the activity content.

That said, it is validating the url and the data at the other end should be small.

Let's discuss with the team.

Reply:
removed